lib/queue.rb

```diff
@@ -1,16 +1,47 @@
 class Queue
 
+  MAX_SIZE = 20
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @queue = []
+    @size = 20
+    @front = -1 
+    @rear = -1
+    
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+
+    if (@front == 0 && @rear == @size - 1) 
+    raise "queue is full"
```

Comment:
Otherwise the Queue could wrap around and the rear could overlap the front. Consider what if you added an element, removed it and then added 25 more elements.

```suggestion
    if ((@front == 0 && @rear == @size - 1)  ||  (@rear == @front && @front != -1) 
      raise "queue is full"
```